github-upload-action-secrets

```diff
@@ -45,10 +51,24 @@ def encrypt(public_key: str, secret_value: str) -> str:
 
     public_key = public.PublicKey(public_key.encode("utf-8"), encoding.Base64Encoder())
     sealed_box = public.SealedBox(public_key)
-    encrypted = sealed_box.encrypt(secret_value.encode("utf-8"))
+    encrypted = sealed_box.encrypt(secret_value)
     return b64encode(encrypted).decode("utf-8")
 
 
+def generate_ssh_key(pubkey):
+    """Generates a Ed25519 keypair and outputs public/private in OpenSSH format"""
+    key = Ed25519PrivateKey.generate()
```

Comment:
so, internally, this is using OpenSSL.

and inside of that, it's calling `getrandom()`.